rise/static/main.js

```diff
@@ -562,25 +618,43 @@ function reveal_cell_index(notebook) {
     return cell.metadata.slideshow
 	&& cell.metadata.slideshow.slide_type == 'subslide';
   }
+  var is_fragment = function(cell) {
+    return cell.metadata.slideshow
+	&& cell.metadata.slideshow.slide_type == 'fragment';
+  }
   var slide_counter = is_slide(notebook.get_cell(0)) ? -1 : 0;
   var subslide_counter = 0;
+  var fragment_counter = 0;    
   var result = null;
 
   notebook.get_cells().forEach(function (cell, index) {
-    if (result)
+    // console.log(`result=${result}, sl=${slide_counter}/${slide} subsl=${subslide_counter}/${subslide} fr=${fragment_counter}/${fragment}`);
+    if (result) {
       // keep it short: skip if we found already
       return;
+    }
     if (is_slide(cell)) {
       slide_counter += 1;
       subslide_counter = 0;
     } else if (is_subslide(cell)) {
       subslide_counter += 1;
     }
     if ((slide_counter == slide) && (subslide_counter == subslide)) {
-	result = index;
+	    // keep count of fragments but only on current slide
+      if (is_fragment(cell)) {
+	      fragment_counter += 1;
+	    }
+      // we're on the right slide
+      // do we need to worry about fragment
+      var in_fragment = (! focus_fragment ) ? true // 
```

Comment:
When `in_fragment` could be false here? Can you explain a little bit more about the whole block? Thanks!